refinery/data_set_manager/views.py

```diff
@@ -1018,3 +1018,41 @@ def post(self, request):
                 import_file.delay(file_store_item.uuid)
 
         return HttpResponse()  # 200
+
+
+class RemoveFileFromNodeView(View):
+    """Remove file from a Node instance"""
+    def post(self, request):
+        try:
+            node = Node.objects.get(uuid=request.POST['node_uuid'])
+        except KeyError:
+            logger.error("Node UUID was not provided in request")
+            return HttpResponseBadRequest()
+        except Node.DoesNotExist:
+            logger.error("Node with UUID '%s' does not exist",
+                         request.POST.get('node_uuid'))
+            return HttpResponseNotFound()
+        except Node.MultipleObjectsReturned:
+            logger.critical("Multiple Nodes found with UUID '%s'",
+                            request.POST.get('node_uuid'))
+            return HttpResponseServerError()
+
+        if request.user != node.study.get_dataset().get_owner():
+            return HttpResponseForbidden()
+
+        logger.debug("Removing file from Node '%s'", node)
+        try:
+            file_store_item = FileStoreItem.objects.get(uuid=node.file_uuid)
+        except FileStoreItem.DoesNotExist:
+            logger.error("FileStoreItem with UUID '%s' does not exist",
+                         node.file_uuid)
+            return HttpResponseNotFound()
+        except FileStoreItem.MultipleObjectsReturned:
+            logger.critical("Multiple FileStoreItems found with UUID '%s'",
+                            node.file_uuid)
+            return HttpResponseServerError()
+
+        file_store_item.delete()
+        node.file_uuid = None
+        node.save()
```

Comment:
How are we handling updating solr, since the file browser grabs it info from assay/files -> solr. Relevant to adding files also.

Reply:
Not sure what needs to happen with Solr. I think it happens automatically in case of adding the files.

Reply:
A reindex would still need to happen here for the Node that is being altered. Otherwise there would still be a link provided in the FileBrowser (populated by the Solr response) pointing to the data file that has been removed here